docs/contributing.rst

```diff
@@ -147,34 +153,68 @@ Some tips for reviewing
 
 
 .. code-block:: python
-
     def znl_mean(ar):
         return ar.mean('xt_ocean')
-
-has a few issues. First, the names don't read English. The method does not have any documentation nor is self-explanatory. Further, the method assumes that ``xt_ocean`` is a coordinate of the data array.
+has a few issues. First, the names aren't easily understood. The method does not have any documentation, nor is it self-explanatory. Furthermore, the method assumes that ``xt_ocean`` is a coordinate of the data array; hard coding dimension names is fragile to future changes.
 
 A much better version, free from all the cons mentioned above, is:
 
 .. code-block:: python
-
     def zonal_mean(dataarray):
         '''
         Returns the (numerical) zonal mean of `dataarray`, i.e., its mean along latitude circles.
-
             Parameters:
                     dataarray (xarray.dataarray): An xarray dataarray
-
             Returns:
                     xarray.dataarray: The (numerical) zonal mean of `dataarray`
         '''
-
         return dataarray.cf.mean('longitude')
-     
-
 * Ensure that the notebook runs! To do that:
 
   - Clone the repository or the fork that the PR was made from;
   - Checkout the appropriate branch;
   - Ensure that the notebook runs when a **new** kernel is launched. Ensure that all cells run in sequential order, and that all cell outputs are evaluated.
 
 .. _existing Pull Requests: https://github.com/COSIMA/cosima-recipes/pulls
+
+
+To clone a pull request locally
++++++++++++++++++++++++++++++++
+
+If you want to test pull requests locally (i.e., to compile or run the code),
+you will need to download the pull request branch. You can do this either by cloning the branch from the pull request.
+
+In this context "locally" means somewhere you can run the code - this is probably on Gadi, but may also be on a local machine.
+
+If you are using ssh keys for command line authentication:
+
+::
+
+    git clone -b «THEIR_DEVELOPMENT_BRANCHNAME» git@github.com:«THEIR_GITHUB_USERNAME»/cosima-recipes.git
+
+where «THEIR_GITHUB_USERNAME» is replaced by the username of the person proposing the pull request,
+and «THEIR_DEVELOPMENT_BRANCHNAME» is the branch from their pull request.
+
+Alternatively, you can add the repository of the user proposing the pull request as a remote to
+your existing local repository. Navigate to your local repository and type
+
+::
+
+    git remote add «THEIR_GITHUB_USERNAME» git@github.com:«THEIR_GITHUB_USERNAME»/cosima-recipes.git
+
+where «THEIR_GITHUB_USERNAME» is replaced by the user name of the person who has made the
+pull request. Then download their pull request changes
+
+::
+
+    git fetch «THEIR_GITHUB_USERNAME»
```

Comment:
why not `git pull ...`?

Reply:
In this case, it's fetching all the refs from the specified remote. You don't (specifically) want to update your currently checked-out branch at this point, since you're trying to check out a branch from the new remote.

Reply:
`git pull ...` will try to merge the changes into your current branch, whereas `git fetch ...` will simply download the changes. If you are reviewing a PR, you almost certainly do not want to merge the changes in to whichever branch you are currently on.